linkedlist/doubly/python/DoublyLinkedList.py

```diff
@@ -0,0 +1,257 @@
+            index += 1
+            current = current.next
+
+        prev = current.previous
+
+        # Set the preceeding node's next pointer to the new node
+        prev.next = newNode
+        # Set new node's previous pointer to the preceeding node
+        newNode.previous = prev
+        # Set the new node's next pointer to the following node
+        newNode.next = current
+        # Set the following node's previous pointer to the new node
+        current.previous = newNode
+
+        self.size += 1
+
+    def removeFirst(self):
+        """
+        The `removeFirst` method removes the head of the linkedlist and returns it.
+
+        :returns removed element
+        """
+
+        # Check if the list is empty
+        if self.head is None:
+            return "List is Empty!"
+
+        # Check if the head element is the only element in the linkedlist
+        if self.head.next is None:
+            self.head = self.tail = None
+            self.size = 0
+            return
+
+        # Removes reference to head element
+        nodeToRemove = self.head
+
+        newHead = self.head.next
+
+        # Set current head's pointer to None
+        nodeToRemove.next = None
+
+        # Set newHead's previous pointer to None
+        nodeToRemove.previous = None
+
+        # Set newHead as head
+        self.head = newHead
+
+        # Decrement size
+        self.size -= 1
+
+        return nodeToRemove.element
+
+    def peek(self):
+        # Check if list is empty
+        if self.head is None:
+            return "Empty List!"
+
+        return self.head.element
+
+    def removeLast(self):
+        """
+        The `removeLast` method removes the tail of the linkedlist.
+
+        :returns the removed element
+        """
+
+        # Check if linkedlist is empty
+        if self.tail is None:
+            return "No tail present in linkedlist!"
+
+        # Check if the tail is the only element in the list
+
+        if self.tail.previous is None:
+            self.head = None
+            self.tail = None
+            self.size = 0
+            return
+
+        nodeToRemove = self.tail
+        newTail = self.tail.previous
+
+        # Set current tail's previous pointer to None
+        nodeToRemove.previous = None
+        # Set the newTail's next pointer to None
+        newTail.next = None
+        # Set the newTail as tail
+        self.tail = newTail
+
+        # Decrement size
+        self.size -= 1
+
+        return nodeToRemove.element
+
+    def removeAtPosition(self, position):
+        """
+        The `removeAtPosition` method removes a node at the specified postion.
+
+        :param position: position to remove element from
+        :returns the removed element
+        """
+
+        # Check if position is valid
+        if position < 1 or position > self.size:
+            raise IndexError("Invalid position!")
+
+        # Check if element to be removed is the head
+        if position == 0:
```

Comment:
Should be if `position == 1` because we are using a 1-indexed approach here. Add a test case that fails because of this in the driver code too, so we are sure it's fixed when you update. @Youngestdev

Reply:
Oh okay!